src/Form/UserPickupRequestForm.php

```diff
@@ -0,0 +1,413 @@
+                // set the expire date for each selected hold
+                // commented out for now during testing
+                // $updated_hold = $guzzle->get("$api_url/patron/$selfCheckApi_key-$patron_barcode/updated_hold/" . $hold['holdId'] . "?shelf_expire_time=$pickup_date 23:59:59")->getBody()->getContents();
+                // create arborcat_patron_pickup_request records for each of the selected holds
+                $db->insert('arborcat_patron_pickup_request')
+                    ->fields([
+                      'requestId' => $holdToRequest['holdId'],
+                      'patronId' => $pnum,
+                      'branch' => (int) $branch,
+                      'timeSlot' => $locationId_timeslot[1],
+                      'pickupLocation' => $locationId_timeslot[0],
+                      'pickupDate' => $pickup_date,
+                      'contactEmail' => ($notification_types['email'] ? $patron_email : null),
+                      'contactSMS' => ($notification_types['sms'] ? $patron_phone : null),
+                      'contactPhone' => ($notification_types['phone'] ? $patron_phone : null),
+                    ])
+                    ->execute();
+            }
+        }
+
+        // Need to add a c"confirm the request" modal dialog here before proceeding
+
+        $user = \Drupal\user\Entity\User::load(\Drupal::currentUser()->id());
+        $uid = $user->id();
+        $url = \Drupal\Core\Url::fromRoute('entity.user.canonical', ['user'=>$user->id()]);
+
+        return $form_state->setRedirectUrl($url);
+    }
+
+    public function arborcat_mail($key, $email_to, $patron, $code, $holds)
+    {
+        if ($key == 'locker_requests') {
+            if ($email_to == 'mcblockers@aadl.org') {
+                $location = 'malletts';
+                $link = \Drupal::config('arborcat.settings')->get('mcb_lockers');
+            } else {
+                // Default to Pitts
+                $location = 'pittsfield';
+                $link = \Drupal::config('arborcat.settings')->get('pts_lockers_insert');
+            }
+            $email_headers = "From: support@aadl.org";
+            $email_subject = 'Locker requested';
+            $email_message = "A patron has requested a locker pickup. Please follow the link below to reserve a locker:\r\n\r\n".$link."\r\n\r\n".
+            "If the locker reservation is successful, then place the requested items in the assigned locker using the below information:\r\n".
+            "Patron name: "."$patron\r\n"."Locker code: ".$code."\r\n\r\nRequested items:\r\n";
+            foreach ($holds as $title) {
+                $email_message = $email_message.$title['Title']."\r\n";
+            }
+            $mailManager = \Drupal::service('plugin.manager.mail');
+            mail($email_to, $email_subject, $email_message, $email_headers);
+        }
+    }
+
+    public function validateForm(array &$form, FormStateInterface $form_state)
```

Comment:
Usually have the validateForm function between the buildForm and the submitForm, just for readability